lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -526,8 +581,8 @@ void SingleExitLoopTransformer::patchPreheader(SILBasicBlock *newHeader) {
   // State from within the loop is not available in the preheader.
   // Simply pass in an undef. This will never be accessed at runtime.
   SmallVector<SILValue, 8> newArgs;
-  for (const SILValue &escapingValue : escapingValues) {
-    newArgs.push_back(getUndef(escapingValue->getType()));
+  for (const auto &kv : escapingValueSubstMap) {
+    newArgs.push_back(kv.second);
```

Comment:
a note for future consideration: if we really think that some undefs cannot be eliminated (may be useful to discuss that first), we may want to log the undefs and/or add stats counter on them

Reply:
No, there should not be any undefs in the final version.